### DISCUSSION TITLE
Request: 'Days until' countdown/'Days since' countup

### DISCUSSION THREAD

**Discussion by fe2o3fish** [simpleavr/s14clock#20] [Ideas]
... in the strings that are cycled through.

Ex.  ~`<12/25>` Days until Christmas
       You've been sober `~>10/11/16<` Days

Replacing the `~<date> `or `~>date<` with the number of days until/since when displayed.
If no years are specified then it's an annual event.

**Comment by simpleavr** [comment]
Thanks for your input, I think it's a good idea.

I had implemented and build a "beta" firmware bin file (bin/firmware_beta.bin).
If you are interested to try, please go to the [download page](https://simpleavr.github.io/s14clock/install.html) and choose the **beta firmware** to download.
Your unit's current config and wifi credentials will be erased as I have to expand the config memory structure for this to work properly.
There are some adjustments to your idea;

. remove '/' seperator to save storage space
. support MMHH, YYMMHH and YYYYMMHH (looks like it can only count back about 50 years due to data type)
. if the greater than, or the less than day not satisfied, the whole content slot will be skipped. i.e. no negative days after christmas

I haven't conclude the documenations and you can see it is very pre-mature, so expect bugs and please report in this thread. Will refine and have it stablize before upping it as a new v2.03


Ex.
```
~<1225? Days until Christmas
~>191231? Days since COVID
```

**Reply by fe2o3fish** [reply]
Thanks!  I've given it a try and have a couple of suggestions:

- If zero or negative day, not displayed but an additional 'cycle time' still occurs.
- Please display if zero.
- If negative, add 365 (366 if (month > 2 and next year is leap year) or (month < 3 and current year is leap year)
  and display.  Otherwise this kinda makes this feature less than useful for dates early in the year.

- Change 'Days behind date' to 'Days after date' in clock's webpage.

Thanks!
        -Rusty-

**Reply by simpleavr** [reply]
If zero or negative day, not displayed but an additional 'cycle time' still occurs.
**Cannot avoid this for now due to current code structure, Will need to add a "dry run" cycle to acheive this which is inefficient. May consider this in the future.**

Please display if zero.
**OK for change**

If negative, add 365 (366 if (month > 2 and next year is leap year) or (month < 3 and current year is leap year)
and display. Otherwise this kinda makes this feature less than useful for dates early in the year.
**This is getting complicated and I don't want to accomodate, I will have a full page write-up to explain how this works. I would probably implement that for ~<MMDD, if date passed, will you next year same date for counts, and for ~>MMDD, if date not yet come, will use lasy year's date**

Change 'Days behind date' to 'Days after date' in clock's webpage.
**Will do, I thrust that's better English :)**

I try to keep S14Clock simple and not specialize to to certain things. For specialization, the code is open sourced and everyone is welcome to clone it and customize for indvidual needs, say turn this into a stock ticker, weather forecast, etc.

**Reply by fe2o3fish** [reply]
Your 'trust' is correct.  😁

**Comment by simpleavr** [comment]
**Changes to days counting tokens**

for `~<MMDD?`, if date passed, will use next year same date for counts
for `~>MMDD?`, if date not yet come, will use last year's date

Change 'Days behind date' to 'Days after date' in clock's webpage

**Add direction override (right to left) for Shift and Drip transitions**
add `~=` in your content entry for shifting / dripping message from right to left, instead of left to right.

Again, please go to the [download page](https://simpleavr.github.io/s14clock/install.html) and choose the beta firmware to download.
Will work on better descriptions and additional explanations in project page later.
